Client authentication

decode9 opened the pull request.

decode9 committed:
added mongoengine to app inizializer
added key generation files
added api_key and create client seeder
added AuthMiddelware and integrate with graphqlRoute
added client model
added too much things

decode9 committed:
key manager util

decode9 committed:
key manager renamed

free1f commented:
Autogenerar carpeta "keys" en caso que no exista, para evitar error "no such file or directory" al levantar la aplicación

decode9 committed:
Added APP_NAME in env
Added APP_NAME in constants
Added app_name in client model
Added app_name in key message
Added app_name validation in key
Added key creation folder
Added key_install file

decode9 committed:
app key verification in signature verify

free1f commented:
Casos de pruebas validados con Docker:
RSA PAIR KEY SERVER GENERATION:
	- Validacion de APP_KEY en el .env [exitosa]
	- Validacion de APP_KEY en el .env al momento de verificar la firma (verify_signature en auth_middleware) [exitosa]
	- Generacion de carpeta keys y archivos .pem
GENERATE CLIENT APP KEY:
	- Validacion de generacion de APP_KEY (evaluar clientes repetidos en la tabla Clients) [exitosa]
	- Retorno de llave en consola [exitosa]
VALIDATE CLIENT APP KEY:
	- Validacion token en el header Api-Key [exitosa]
	- Validacion token en el header Api-Key / Cliente existe[exitosa]
	- Validacion token en el header Api-Key / Cliente inactivo [exitosa]

decode9 committed:
unique client name
